azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -74,10 +81,18 @@ protected void handleGet(
       throws ServletException, IOException {
     try {
       final Map<String, String> templateVariableToValue = new HashMap<>();
-      if (IMAGE_TYPE_WITH_ID_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
-        // TODO: Implementation will be provided in the future PR
-      } else if (GET_IMAGE_TYPE_URI.equals(req.getRequestURI())) {
-        // TODO: Implementation will be provided in the future PR
+      ImageTypeDTO imageTypeDTO;
+      if (IMAGE_TYPE_WITH_NAME_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
+        String imageTypeName = templateVariableToValue.get(IMAGE_TYPE_NAME_KEY);
+        if (!hasImageManagementPermission(imageTypeName, session.getUser(), Type.GET)) {
+          log.info(FORBIDDEN_USER_ERR_MSG);
+          throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN, FORBIDDEN_USER_ERR_MSG);
+        }
+        imageTypeDTO = this.imageTypeService.findImageTypeWithOwnersByName(imageTypeName);
+        sendResponse(resp, HttpServletResponse.SC_OK, imageTypeDTO);
+      } else {
+        log.info(PATH_NOT_SUPPORTED);
+        throw new ImageMgmtInvalidInputException(PATH_NOT_SUPPORTED);
       }
     } catch (final Exception e) {
```

Comment:
If an authorization error occurred we should return status code 403, for internal server errors the correct code is 500. However, the current implementation will always return 404 regardless of the nature of the error which is misleading. Could you please fix this?

Reply:
That's a very good point, yes.